### ISSUE TITLE
3+ players

### ISSUE THREAD

**Issue by kethinov** [mcclure/emu-coop#4]
This project is cool!

I recently tried connecting 3 players like this:

- A connects to B
- B connects to C
- C connects to A

The intention of course is to get all 3 players to share inventory.

But it appears the script is not designed to work that way.

Dunno how hard this would be to implement, but I thought it would be a neat idea so I'll put it up here.

In the mean time, I really enjoyed a couple 2 player play throughs. This script breathes new life into Zelda 3! 😀

**Comment by Xkeeper0** [comment]
The way it works is that it communicates directly to another client over IRC. In theory, multi-user play could be achieved by joining a (private) channel instead and using messages from it rather than one username. This would allow anybody to jump in / drop out, but of course only the changes going forward would be reported...

**Comment by kethinov** [comment]
Yeah, that makes sense. I suppose it would require tweaking how the connection UI works to enable a "one to many" approach on a private server as opposed to explicitly connecting to a specific partner.

To mitigate the "only changes going forward" problem, perhaps support for matrix could be added as an alternative to IRC, as matrix supports server-side history.

**Comment by mcclure** [comment]
Doing it as a channel in IRC would be very easy.

If we wrote and hosted a server, that would also make things easy.

**Comment by adan-ferguson** [comment]
Doing it in an IRC channel would also help with the issue of lingering nicknames, since you wouldn't really need a specific one anymore
